consumption.tex

```diff
@@ -0,0 +1,74 @@
+\section{Consumption}
+
+We now describe how a smart contract on the target blockchain can verify that a burn has taken place on the source blockchain. Note how this puts a constraint on the target blockchain to have smart contract capabilities. However there is no such constraint for the source blockchain. Such a mechanism allows a target cryptocurrency to bootstrap through burns on another source blockchain as outlined in Section~\ref{section:introduction}.
+
+A user who wishes to acquire a target cryptocurrency first has to form a burn address which encodes her receiving address on the target blockchain by using it as a tag. The user then sends an amount of the source cryptocurrency to that address. She then proceeds to submit a proof of this burn to a smart contract on the target blockchain, where the burn is verified and the user is credited an equivalent amount of target cryptocurrency on her receiving address.
+
+In accordance to the terminology laid out in~\cite{pow-sidechains}, we call this party the \emph{prover} and we call the smart contract the \emph{verifier}.
+
+We abstract the functionalities regarding transactions on the source blockchain into a \emph{blockchain transaction protocol}.
+
+\begin{definition}[Blockchain transaction protocol]\label{def:blockchain-tx}
+  A \emph{blockchain transaction protocol} $\Pi_\zeta$ consists of two functions $\VerTxIncl$ and $\TxHasTransfer$.
```

Comment:
I'd prefer to drop this definition (i.e., not name this protocol a "blockchain transaction protocol") and just say something along the lines of: "We assume that the source blockchain provides the following two functions which can be written in the smart contract language of the target blockchain:" and then state the functions without a \begin{definition}.